python/rpdk/java/templates/init/guided_aws/Translator.java

```diff
@@ -0,0 +1,114 @@
+    return awsRequest;
+  }
+
+  /**
+   * Translates resource object from sdk into a resource model
+   * @param awsResponse the aws service describe resource response
+   * @return model resource model
+   */
+  static ResourceModel translateFromReadResponse(final AwsResponse awsResponse) {
+    // e.g. https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/blob/master/aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/Translator.java#L58
+    return ResourceModel.builder()
+        //.primaryIdentifier(response.primaryIdentifier())
+        //.someProperty(response.property())
+        .build();
+  }
+
+  /**
+   * Request to delete a resource
+   * @param model resource model
+   * @return awsRequest the aws service request to delete a resource
+   */
+  static AwsRequest translateToDeleteRequest(final ResourceModel model) {
+    final AwsRequest awsRequest = null;
+    // TODO: construct a request
+    // e.g. https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/blob/master/aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/Translator.java#L33
+    return awsRequest;
+  }
+
+  /**
+   * Request to update properties of a previously created resource
+   * @param model resource model
+   * @return awsRequest the aws service request to modify a resource
+   */
+  static AwsRequest translateToUpdateRequest(final ResourceModel model) {
+    final AwsRequest awsRequest = null;
+    // TODO: construct a request
+    // e.g. https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/blob/master/aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/Translator.java#L45
+    return awsRequest;
+  }
+
+  /**
+   * Request to update properties of a previously created resource
+   * @param nextToken token passed to the aws service describe resource request
+   * @return awsRequest the aws service request to describe resources within aws account
+   */
+  static AwsRequest translateToListRequest(final String nextToken) {
+    final AwsRequest awsRequest = null;
+    // TODO: construct a request
+    // e.g. https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/blob/master/aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/Translator.java#L26
+    return awsRequest;
+  }
+
+  /**
+   * Translates resource objects from sdk into a resource model (primary identifier only)
+   * @param awsResponse the aws service describe resource response
+   * @return list of resource models
+   */
+  static List<ResourceModel> translateFromListRequest(final AwsResponse awsResponse) {
+    // e.g. e.g. https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/blob/master/aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/Translator.java#L81
+    return streamOfOrEmpty(Lists.newArrayList())
+        .map(resource -> ResourceModel.builder()
```

Comment:
I thought List is meant to return a list of primaryIdentifiers, not a list of resource models. perhaps this has not been updated across all the packages/interfaces...